### ISSUE TITLE
When Hint.ThreadSafe is set to "Off" Source generator creates a lock in Dispose method

### ISSUE THREAD

**Issue by tahacorbaci** [DevTeam/Pure.DI#81]
When Hint.ThreadSafe is set to "Off", the lock is expected to be unused, but in the Dispose method, the source generator writes and tries to use a lock that was not generated.

Version: 2.1.47

```
internal partial class Composition
{
    private void Setup()
    {
        DI.Setup("Composition")
            .Hint(Hint.Resolve, "Off")
            .Hint(Hint.ThreadSafe, "Off")
            .DefaultLifetime(Lifetime.Singleton)
            .Bind<IFoo>().To<Foo>()
            .Root<IFoo>("Foo");
    }
}

internal class Foo : IFoo
{
    private readonly Dependency _dependency;

    public Foo(Dependency dependency)
    {
        _dependency = dependency;
    }

    public void Dispose() { }
}

internal interface IFoo : IDisposable { }

internal class Dependency : IDependency { }

internal interface IDependency { }
```

In generated code _lockM12D29di is not created but used in Dispose method
```
public void Dispose()
{
  int disposeIndex;
  object[] disposables;
  using (_lockM12D29di.EnterScope())
  {
	  disposeIndex = _disposeIndexM12D29di;
	  _disposeIndexM12D29di = 0;
	  disposables = _disposablesM12D29di;
	  _disposablesM12D29di = new object[1];
	  _singletonFooM12D29di41 = null;
  }
  
  while (disposeIndex-- > 0)
  {
	  switch (disposables[disposeIndex])
	  {
		  case global::System.IDisposable disposableInstance:
			  try
			  {
				  disposableInstance.Dispose();
			  }
			  catch (Exception exception)
			  {
				  OnDisposeException(disposableInstance, exception);
			  }
			  break;
	  }
  }
}
```

**Comment by NikolayPianikov** [comment]
@tahacorbaci thanks for the bug report

**Comment by NikolayPianikov** [comment]
Fixed in 2.1.48
